Improve the readability of SCDoc TOC

prko opened the pull request:
<!-- For information about contributing see: https://github.com/supercollider/supercollider/wiki/Contributing-directory -->

## Purpose and Motivation
In some documentation, the current table of contents (TOC) may be confusing for users due to its layout. At times, this could even be perceived as a bug.
As demonstrated in the two comparison examples below, this pull request (PR) enhances the readability of the SCDoc TOC:
![Screenshot 2025-05-23 at 09 44 41](https://github.com/user-attachments/assets/d6927f78-eb5f-408d-98ff-11056636589d)![Screenshot 2025-05-23 at 09 44 06](https://github.com/user-attachments/assets/269db9d3-c2b9-46de-8652-9e40abfe3b6f)
![Screenshot 2025-05-23 at 09 47 23](https://github.com/user-attachments/assets/bfc96301-6910-4c76-9cbc-9185d0bd197c)![Screenshot 2025-05-23 at 09 47 03](https://github.com/user-attachments/assets/cbdfee9a-ce57-42d9-a215-53ab0213a60b)

<!-- Please provide a description, even if you are linking to a related Issue or PR. -->
<!-- If this fixes an open issue, link to it by writing "Fixes #555." -->

## Types of changes

<!-- Delete lines that don't apply -->

- Bug fix
- New feature

## To-do list

<!-- Complete an item by checking it: [x]. Add new entries to track your progress -->

- [x] Code is tested
- [x] All tests are passing
- [x] This PR is ready for review <!-- If not ready for review, consider making a Draft PR instead, or describe what kind of specific feedback you want. -->

<!-- ## Merge notes
(optional) You may also add notes of:
     - Dependencies: if this PR depends on any other PRs or actions
     - Merge Note: any considerations regarding merging this PR
     - Release Notes: text you think would be useful to include in the Release Notes
-->

<!-- Note: The recommended commit message format is:
     component: sub-component: short description of changes
For example:
     classlib: ClassBrowser: fix search with empty query string
Common `comp:` labels:
     help, classlib, sclang, scsynth, plugins, tests, docs, SCDoc
See wiki for more info: Wiki->Creating Pull Requests->git-commit-message-format
-->

capital-G commented:
IMO it looks clearer w/o the ticks? I think this is a rather subjective change, I don't know how others feel about it?

The technical implementation should be done differently via https://developer.mozilla.org/en-US/docs/Web/CSS/list-style though - don't inject bullets as text.
I had a look at this and it seems the styling is not as easy as it should be because the SCDoc HTML has invalid syntax regarding the TOC (e.g. we open a `ul` an place another `ul` into it, although this nested `ul` should be wrapped in a `li` - but there are many other places as well, e.g. `<p/>`) - the browser recovers from this syntax but we can't style this properly anymore.
I'd personally advise to abandon SCDoc in its current state and focus our efforts more on a complete rewrite of SCDoc for 3.15.

prko commented:
>I think this is a rather subjective change, 

It could be because this is my latest feeling.
It did not bother me before... Strange...

Let's see what others think!

prko commented:
@capital-G @tedmoore 
Thanks for your valuable opinions!

- I removed adding `- ` before each title. The reason is as follows:
I tested it with my tutorials for my students. In my tutorials, I use a numbered title as follows:
   ```
   1. An introductory overview
   2. Getting started
   2.1. Getting Sound
   2.1.1. Generating Sound from Scratch
   2.1.2. Defining and Instantiating Synths
   2.1.3. Live Coding Essentials
   2.1.4. Building Musical Patterns
   ```
   With this style, `- ` decreases the readability.

- Instead, I adjusted the line padding between toc items.

- I think adding `.` before method names looks good.

The following is an example of the rendering using the last commit:   

I would be happy if the development team considered this pull request when revising the approach to SCDoc. For now, I will leave this PR as open.
![Screenshot 2025-06-11 at 11 20 39](https://github.com/user-attachments/assets/fec47f93-13a9-4abd-9cc8-529251623e15)
![Screenshot 2025-06-11 at 11 20 54](https://github.com/user-attachments/assets/c4527082-1edf-4d30-b094-3d0e7d8d1d6b)
